contrib/docker/docker-compose-postgres/docker-compose.yaml

```diff
@@ -0,0 +1,38 @@
+version: '3.4'
+services:
+  postgres:
+    image: postgres:11-alpine
+    container_name: postgres
+    restart: always
+    ports:
+      - '5432:5432'
+    environment:
+      POSTGRES_USER: backstage
+      POSTGRES_PASSWORD: backstage
+    volumes:
+      - ./tools/development/db/:/docker-entrypoint-initdb.d
+  backstage:
+    image: node:12
+    container_name: backstage
+    working_dir: /workdir
+    command: bash -c "yarn install && yarn dev"
+    tty: true
+    environment:
+      NODE_ENV: development
+      POSTGRES_HOST: postgres
+      POSTGRES_USER: backstage
+      POSTGRES_PASSWORD: backstage
+    depends_on:
+      - postgres
+    ports:
+      # App server
+      - '3000:3000'
+      # Backend server port
+      - '7000:7000'
+      # Backend Node debug port
+      - '9229:9229'
+    volumes:
+      - .:/workdir
+      - node_modules:/workdir/node_modules
```

Comment:
There are more node_modules folders than this one, one for every package. Maybe it's best to just not try to separate it from the host ones, even if it means the project won't be runnable on the host at the same time in case of OS incompatibilities?

Reply:
The reason why it was like this in the first place is because Docker for Mac is incredibly slow when traversing between host/Docker. There are multiple, but this one is the largest one. The way the project is set up currently would make setting up one line per copy to be very difficult.

Reply:
Will this actually work correctly in the backend if you include just the root node_modules? Yarn workspaces doesn't always install all packages in the root (or bin commands) and the backend doesn't currently bundle them. So I could see this missing some of the required packages. I could see this causing runtime errors in some cases.

Reply:
Oh wait I see your doing a yarn install command on startup. That is probably what is working around the issue I describe here.

Reply:
I think worst case the integrity check might actually succeed and cause none of the packages to get any `node_modules` installed at all. Not sure exactly what yarn checks, but there at least aren't any integrity files in the nested `node_modules`..

Reply:
Yeah this could actually cause unwanted changes in the host system node_modules by running yarn instal in the container while it's volume mounted. That and some packages that install native libraries won't work properly. This has also been a frustrating part of trying to volume mount node_modules.

I'm not aware of a great solution here except always install all node_modules inside docker. If you avoid mounting node_modules this avoids these pitfalls but of course slows down startup.

Reply:
Agree with all of these - I think we'd always be trading the dev experience for correctness. I've got a fancy setup that uses rsync and fswatch -- but I don't think many people would take the trouble to do something that sophisticated.

Having a node_modules used by both host and container will always be a problem, so I don't think that's something we would want to support. I do observe that all node_modules I have are empty, unless I explicitly mark a package as `nohoist`.

If it's not possible to have a generic solution then maybe it may not be the best idea to support this, or maybe state the scenarios in which this solution doesn't work.

Reply:
I tend to find my workflow still running node locally as such.

1. Start postgres `docker-compose up -d postgres`
2. Start local dev `yarn dev` pointing databse to `localhost` in `app-config.yml`

Easy to get up and running plus the benefits of running Node locally, compile time speed and node_modules inspection.  

This could the recommended docker workflow. Running Node in Docker doesn't seem as paramount. Most devs will already have Node installed locally at least for IDE support.

Reply:
@gabemeola from my experience it's always been a pain to have a development environment within docker. I rely, like you, for docker to provide dependencies rather than my dev environment. It's just easier that way I think